src/components/VaultRow/lib/VaultDepositForm.tsx

```diff
@@ -34,31 +32,39 @@ import {
   InvalidMinAmountModal,
 } from "@/components/Modal"
 import TokenForm, { TokenFormValues } from "@/components/TokenForm/TokenForm"
+import { VaultRowPropsWithProduct } from "@/components/VaultRow/VaultRow"
 
 import { useGlobalStore } from "@/store"
 
-export const VaultDepositForm: FC<VaultProps> = (props) => {
+export type VaultDepositWithdrawProps = VaultRowPropsWithProduct & {
+  inputToken: Address
+  outputToken: Address
+  underlyingAssets: Address[] | readonly Address[] | undefined
```

Comment:
Just use `?:` when you want to allow a type to be undefined

Reply:
Same here, what's the rule?